src/enclave/ServiceProvider/ServiceProvider.cpp

```
@@ -147,284 +151,197 @@ void ServiceProvider::export_public_key_code(const std::string &filename) {
   file.close();
 }
 
-void ServiceProvider::connect_to_ias(const std::string &ias_report_signing_ca_file) {
-  try {
-    connect_to_ias_helper(ias_report_signing_ca_file);
-  } catch (const std::runtime_error &e) {
-    if (require_attestation) {
-      throw;
-    }
+bool verify_mrsigner(char* signing_public_key_buf,
+                     size_t signing_public_key_buf_size,
+                     uint8_t* signer_id_buf,
+                     size_t signer_id_buf_size) {
+  
+  mbedtls_pk_context ctx;
+  mbedtls_pk_type_t pk_type;
+  mbedtls_rsa_context* rsa_ctx = NULL;
+  uint8_t* modulus = NULL;
+  size_t modulus_size = 0;
+  int res = 0;
+  bool ret = false;
+  unsigned char* signer = NULL;
+
+
+  signer = (unsigned char*)malloc(signer_id_buf_size);
+  if (signer == NULL) {
+    printf("Out of memory\n");
+    goto exit;
   }
-}
 
-void ServiceProvider::connect_to_ias_helper(const std::string &ias_report_signing_ca_file) {
-  if (this->ias) {
-    return;
-  }
+  mbedtls_pk_init(&ctx);
 
-  std::unique_ptr<IAS_Connection> ias(
-    new IAS_Connection(is_production ? IAS_SERVER_PRODUCTION : IAS_SERVER_DEVELOPMENT, 0));
+  res = mbedtls_pk_parse_public_key(&ctx,
+                                    (const unsigned char*)signing_public_key_buf,
+                                    signing_public_key_buf_size+1);
 
-  const char *ias_client_cert_file = std::getenv("IAS_CLIENT_CERT_FILE");
-  if (!ias_client_cert_file) {
-    throw std::runtime_error(
-      "Set $IAS_CLIENT_CERT_FILE to the file generated by openssl x509, probably called "
-      "${OPAQUE_HOME}/client.crt.");
+  if (res != 0) {
+    printf("mbedtls_pk_parse_public_key failed with %d\n", res);
```

Instead of printf can we instead throw a runtime error?